Makefile

```diff
@@ -256,14 +256,18 @@ status:
 
 checkout:
 	@for r in $(REPOS) ; do \
-	  cd $$r; \
-	  ((git -c 'color.ui=always' checkout $(branch) 2>&1 | \
-	   sed "s|^|$$r: |") || \
-	  (git -c 'color.ui=always' checkout $(DEFAULT_BRANCH) 2>&1 | \
-	   sed "s|^|$$r: |")) && \
-	  git pull | sed "s|^|$$r: |"; \
-	  echo; \
-	  done
+		cd $$r; \
+		git show-ref --verify --quiet refs/heads/$(branch); \
+		if [ $$? -eq 0 ]; then \
+			git -c 'color.ui=always' checkout $(branch) 2>&1 | sed "s|^|$$r: |"; \
+			git pull | sed "s|^|$$r: |"; \
+		else \
+			echo "$(branch) doesn't exist, checking out $(DEFAULT_BRANCH)..."; \
+			git -c 'color.ui=always' checkout $(DEFAULT_BRANCH) 2>&1 | sed "s|^|$$r: |"; \
+			git pull | sed "s|^|$(DEFAULT_BRANCH): |"; \
```

Comment:
Is the `sed` recipe correct here. In the above block we're printing the repo name at the beginning of the line; here we're printing the branch name.

Along those lines, it would be nice if the repetition of lines 262+263 / 266+267 could be eliminated.

Reply:
good catch! fixed and DRYed up